### PR TITLE
Paragon table deprecation updation

### PR DESCRIPTION
**Pull request by abdullahwaheed**
### Ticket
[Migrate off deprecated Paragon components](https://github.com/openedx/frontend-wg/issues/102)

### What has changed
Removed deprecated `Table` component of paragon and updated it with `DataTable`

### References
[Paragon Table](https://paragon-openedx.netlify.app/components/table/)
[Paragon DataTable](https://paragon-openedx.netlify.app/components/datatable/)

### CODE REVIEW

File: src/components/GradesView/EditModal/OverrideTable/__snapshots__/test.jsx.snap [hunk 1]
```diff
@@ -1,40 +1,40 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`OverrideTable Component snapshots basic snapshot shows a row for each entry and one editable row 1`] = `
-<Table
+<Component
```
Comment (muselesscreator) [inline src/components/GradesView/EditModal/OverrideTable/__snapshots__/test.jsx.snap:4]: Gotta add DataTable to the setupTest script mocks

File: src/components/GradesView/GradebookTable/__snapshots__/test.jsx.snap [hunk 2]
```diff
@@ -7,32 +7,38 @@ exports[`GradebookTable component snapshot - fields1 and 2 between email and tot
   <div
     className="gbook"
   >
-    <Table
+    <DataTable
+      EmptyTableComponent={[Function]}
```
Comment (muselesscreator) [inline src/components/GradesView/GradebookTable/__snapshots__/test.jsx.snap:11]: Gotta mock these components

**Review by davidjoy — APPROVED** [review]
This seems like a fine change - the owning team should also review so they're aware of the change, it should be tested by Fed-BOM and we can get it in.  I did a cursory review.

**Comment by leangseu-edx** [comment]
My apology for add 4 commits to this pr. I was having trouble starting the process. I had to fix a couple of **unrelated bugs** for this PR. I would like like to go through the UI with @muselesscreator before approving this pr.

Screen Shot 2022-06-28 at 11 50 03 AM

**Comment by abdullahwaheed** [comment]
> My apology for add 4 commits to this pr. I was having trouble starting the process. I had to fix a couple of **unrelated bugs** for this PR. I would like like to go through the UI with @muselesscreator before approving this pr.
> 
> Screen Shot 2022-06-28 at 11 50 03 AM

We can use `DataTable.Table` which will use legacy `Table` and it won't add Items count in the table

File: src/components/BulkManagementHistoryView/__snapshots__/HistoryTable.test.jsx.snap [hunk 3]
```diff
@@ -42,78 +42,107 @@ Array [
...
-        },
-        Object {
-          "columnSortable": false,
-          "key": "timeUploaded",
-          "label": "When",
-          "width": "col",
-        },
-      ]
+<DataTable
+  EmptyTableComponent={[Function]}
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/components/BulkManagementHistoryView/__snapshots__/HistoryTable.test.jsx.snap:46]: add DataTable to the setupTests file plz

**Comment by muselesscreator** [comment]
If the new table includes the summary display, we should remove our custom one

**Comment by muselesscreator** [comment]
aaaaactually... we need to hide the paragon one, as we have filters that are not taken into account in the datatable's filters (course members, etc)

**Comment by muselesscreator** [comment]
I have a fix for the row status as well as test updates. Would you be comfortable with me pushing them into this branch before merge? @abdullahwaheed

**Review by muselesscreator — APPROVED** [review]

**Comment by muselesscreator** [comment]
image

**Pull request merged**